src/lib/helpers.ts

```diff
@@ -340,3 +340,12 @@ export function replaceString(str: string, stringA: string, stringB: string) {
   const regExp = new RegExp(escapedStringA, "g");
   return str.replace(regExp, stringB);
 }
+
+// getSubstringAfterLastColon for getting the controller image
+export function getSubstringAfterLastColon(input: string) {
+  const lastColonIndex = input.lastIndexOf(":");
+  if (lastColonIndex === -1) {
+    return "latest";
+  }
+  return input.substring(lastColonIndex + 1);
+}
```

Comment:
Maybe add a check here to make sure you are getting a version number and not a port number?